video_store.py

```diff
@@ -0,0 +1,147 @@
+            postal_code = self.selected_customer["postal_code"]
+        if not phone:
+            phone = self.selected_customer["phone_number"]
+        
+        relevant_params = {
+            "name": name,
+            "postal_code": postal_code,
+            "phone": phone
+        }
+
+        response = requests.put(
+            self.url + f"/customers/{self.selected_customer['id']}",
+            json=relevant_params)
+        self.selected_customer = response.json()
+        print("Here's what we have now: ", self.selected_customer) # self.selected_customer = {'id': 198, 'name': 'Maxine Waters', 'phone': '222-333-4444', 
+                                                                    # 'postal_code': '90211', 'registered_at': 'Sat, 29 May 2021 01:56:21 GMT',
+                                                                    # 'videos_checked_out_count': 0}
+        return response.json()
+
+    def update_rental(self, due_date=None): # Work on rental instance, not video(?)
+        if not due_date:
+            due_date = (self.selected_rental["check_out_date"] + (datetime.timedelta(days=7)))
+        relevant_param = {"due_date": due_date}
+        response = requests.put(self.url + f"/customers/{self.selected_customer['id']}/rentals", # issue is here: TypeError: 'NoneType' object is not subscriptable
+                                json=relevant_param)
+        # can't see any of this bc path above is wrong
+        print("Here's what we have now: ", response)
+        self.selected_rental = response.json()
+        return response.json()
+
+    def delete_customer(self):
+        response = requests.delete(self.url + f"/customers/{self.selected_customer['id']}")
+        self.selected_customer = None
+        return response.json()
+
+    def delete_video(self):
+        response = requests.delete(self.url + f"/videos/{self.selected_video['id']}")
+        self.selected_video = None
+        return response.json()
+
+    def show_selected_customer(self):
+        if self.selected_customer:
+            print(f"Customer with ID {self.selected_customer['id']} is currently selected\n")
+
+    def show_selected_video(self): 
+        if self.selected_video:
+            print(f"Video with ID {self.selected_video['id']} is currently selected\n")
+    
+    # incomplete/incorrect logic 5/27/21, 9:03pm
+    def check_out(self, customer_id=None, video_id=None, check_out_date=None):
+        relevant_params = {
+            "customer_id": customer_id,
+            "video_id": video_id,
+            "check_out_date": check_out_date
+        }
+        if self.selected_customer == None:
+            return "Could not find a customer by that name or ID."
+        if self.selected_video == None:
+            return "Could not find a video by that title or ID."
+        response = requests.post(self.url + "/rentals/check-out", json=relevant_params)
+        return response.json()
+
+    def check_in(self):
+        pass
```

Comment:
Check in winds up looking very similar to check out:

```suggestion
    def check_in(self):
        if self.selected_customer == None:
            return "Could not find a customer by that name or ID."
        if self.selected_video == None:
            return "Could not find a video by that title or ID."

        relevant_params = {
            "customer_id": self.selected_customer["id"],
            "video_id": self.selected_video["id"],
        }
        response = requests.post(self.url + "/rentals/check-in", json=relevant_params)
        return response.json()
```